eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -88,14 +88,17 @@ case class Commitments(channelVersion: ChannelVersion,
     // we need to base the next current commitment on the last sig we sent, even if we didn't yet receive their revocation
     val remoteCommit1 = remoteNextCommitInfo.left.toOption.map(_.nextRemoteCommit).getOrElse(remoteCommit)
     val reduced = CommitmentSpec.reduce(remoteCommit1.spec, remoteChanges.acked, localChanges.proposed)
+    // The funder always pays the on-chain fees, so we must subtract that from the amount we can send.
     val feesMsat = if (localParams.isFunder) commitTxFee(remoteParams.dustLimit, reduced).toMilliSatoshi else 0.msat
-    (reduced.toRemote - remoteParams.channelReserve.toMilliSatoshi - feesMsat).max(0 msat)
+    // We also need to take into account the fact that adding an htlc increases the commit tx fee (regardless of which side is funding).
+    (reduced.toRemote - remoteParams.channelReserve.toMilliSatoshi - feesMsat - htlcFee(reduced.feeratePerKw)).max(0 msat)
```

Comment:
`htlcFee` should only be paid by the funder.

Also depending on the amount available, the htlc may be trimmed and there wouldn't be any htlc fee

Reply:
Good point, I fixed that and I think this resolves the questions I add (I was confusing myself there, with the latest changes it should be what we want).